sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageBlobContainersTests.java

```diff
@@ -100,4 +107,45 @@ public void canUpdateBlobContainer() {
         Assertions.assertEquals(PublicAccess.BLOB, blobContainer.publicAccess());
         Assertions.assertEquals(metadataTest, blobContainer.metadata());
     }
+
+    @Test
+    @Disabled("Need to set env CLI_USERNAME for CLI authentication, e.g. johndoe@microsoft.com")
+    public void canUsePipelineInDataPlane() {
+        String userName = Configuration.getGlobalConfiguration().get("CLI_USERNAME");
+        Assertions.assertNotNull(userName);
+
+        String saName = generateRandomResourceName("javacmsa", 15);
+        String containerName = "blob-test";
+        StorageAccount storageAccount =
+            storageManager
+                .storageAccounts()
+                .define(saName)
+                .withRegion(Region.US_EAST)
+                .withNewResourceGroup(rgName)
+                .create();
+
+        BlobContainers blobContainers = this.storageManager.blobContainers();
+        BlobContainer blobContainer =
+            blobContainers
+                .defineContainer(containerName)
+                .withExistingStorageAccount(rgName, saName)
+                .withPublicAccess(PublicAccess.CONTAINER)
+                .create();
+
+        // assign data-plane blob role
+        msiManager.authorizationManager().roleAssignments()
+            .define(UUID.randomUUID().toString())
+            .forUser(userName)
+            .withBuiltInRole(BuiltInRole.STORAGE_BLOB_DATA_CONTRIBUTOR)
+            .withScope(blobContainer.id())
+            .create();
```

Comment:
I'm surprised that the `create` is not defined as LRO in swagger, though it can still return 201. In my test, service occasionally returns 201 and test fails due to role assignments not created. 

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/stable/2022-04-01/authorization-RoleAssignmentsCalls.json#L235

Reply:
Let us do test now. May need to raise to service if we have outside report.